File: Selection-Sort/selection_sort.js
function findSmallest(array) {
    let minVal = array[0]
    let minIdx = 0

    for (let i = 0; i < array.length; i++) {
        if (array[i] < minVal) {
            minVal = array[i]
            minIdx = i
        }
    }
    return minIdx
}

// Difficulty: log(x^2)
function selectionSortAsc(array) {
    const len = array.length
    const sortedArr = []

    for (let i = 0; i < len; i++) {
        const smallestIdx = findSmallest(array)
        sortedArr.push(...array.splice(smallestIdx, 1))
    }
    return sortedArr
}

module.exports = selectionSortAsc

File: selection_sort/selection_sort.go
package selection_sort

// Difficulty: O(n*2)
func SelectionSortAsc(arr []int) {
	var ln = len(arr)
	for i := 0; i < ln; i++ {
		var minIdx = i
		for j := i; j < ln; j++ {
			if arr[j] < arr[minIdx] {
				minIdx = j
			}
		}

		arr[i], arr[minIdx] = arr[minIdx], arr[i]
	}
}
